src/lib/mac-address.ts

```diff
@@ -0,0 +1,28 @@
+import * as fs from 'mz/fs';
+import * as path from 'path';
+
+export class MacAddressError extends Error {}
+
+export async function getAll(sysClassNet: string): Promise<string | undefined> {
+	try {
+		// does the directory exist?
+		if (!(await fs.exists(sysClassNet))) {
+			return;
+		}
+
+		// read the directories in the sysfs...
+		const interfaces = await fs.readdir(sysClassNet);
+		const addresses = await (
+			await Promise.all(
+				interfaces
+					.map((intf) => path.join(sysClassNet, intf, 'address'))
+					.filter((addr) => fs.existsSync(addr))
```

Comment:
I think typically the right thing to do here is attempt to read it without checking that it exists. Additionally, please don't use any `Sync` functions under normal operation.

Reply:
My concern here was avoiding this scenario in the `Promise.all`:
> It rejects with the reason of the first promise that rejects

I could wrap the inner in a try/catch too but felt this might be less readable/messy. I will try and find some middle ground.